mongo/client_encryption.go

```diff
@@ -71,9 +72,70 @@ func NewClientEncryption(keyVaultClient *Client, opts ...*options.ClientEncrypti
 	return ce, nil
 }
 
-// CreateDataKey creates a new key document and inserts it into the key vault collection. Returns the _id of the
-// created document.
-func (ce *ClientEncryption) CreateDataKey(ctx context.Context, kmsProvider string, opts ...*options.DataKeyOptions) (primitive.Binary, error) {
+// clientEncryptionIDQuery will take a primitive.Binary representation of a CLE UUID and write it as the bytes for a
+// bsoncore.Document.
+func clientEncryptionIDQuery(id primitive.Binary) []byte {
+	return bsoncore.NewDocumentBuilder().AppendBinary("_id", id.Subtype, id.Data).Build()
+}
+
+type ceSingleResultExecutor func(*Collection, []byte) *SingleResult
+
+// executeCESingleResult wraps CLE data to execute some id-specific query operation.
+func executeCESingleResult(ce *ClientEncryption, id *primitive.Binary, fn ceSingleResultExecutor) *SingleResult {
```

Comment:
Clever logic here. Took me a second to read, but I think this pattern with `ce*Executor`s is a good way to reduce duplication. What do you think @kevinAlbs ?